service/storage_service.cc

```diff
@@ -778,7 +778,7 @@ storage_service::get_range_to_address_map_in_local_dc(
     for (auto entry : orig_map) {
         auto& addresses = filtered_map[entry.first];
         addresses.reserve(entry.second.size());
-        std::copy_if(entry.second.begin(), entry.second.end(), std::back_inserter(addresses), local_dc_filter);
+        std::copy_if(entry.second.begin(), entry.second.end(), std::back_inserter(addresses), std::cref(local_dc_filter));
```

Comment:
How does `cref()` help here? Won't the lambda just be moved normally?

Reply:
Note that it's `local_dc_filter` (a local variable), not `get_local_dc_filter()` (a prvalue). `std::copy_if` and other `<algorithm>` stuffs take their function argument by value. The `cref` makes this value a reference to, rather than a copy of the lambda. The lambda contains a string in its capture, so preventing a copy is worthwhile.

There are many other similar uses of variable-bound lambdas/std::functions (including `local_dc_filter`) throughout the code and they all use reference wrappers for that reason. This instance was simply overlooked.